src/queries/label-text.js

```diff
@@ -45,74 +69,69 @@ function queryAllByLabelText(
 ) {
   checkContainerType(container)
 
+  const matcher = exact ? matches : fuzzyMatches
   const matchNormalizer = makeNormalizer({collapseWhitespace, trim, normalizer})
-  const labels = queryAllLabelsByText(container, text, {
-    exact,
-    normalizer: matchNormalizer,
-  })
-  const labelledElements = labels
-    .reduce((matchedElements, label) => {
-      const elementsForLabel = []
-      if (label.control) {
-        elementsForLabel.push(label.control)
-      }
-      /* istanbul ignore if */
-      if (label.getAttribute('for')) {
-        // we're using this notation because with the # selector we would have to escape special characters e.g. user.name
-        // see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Escaping_special_characters
-        // <label for="someId">text</label><input id="someId" />
-
-        // .control support has landed in jsdom (https://github.com/jsdom/jsdom/issues/2175)
-        elementsForLabel.push(
-          container.querySelector(`[id="${label.getAttribute('for')}"]`),
-        )
-      }
-      if (label.getAttribute('id')) {
-        // <label id="someId">text</label><input aria-labelledby="someId" />
-        Array.from(
-          container.querySelectorAll(
-            `[aria-labelledby~="${label.getAttribute('id')}"]`,
-          ),
-        ).forEach(element => elementsForLabel.push(element))
-      }
-      if (label.childNodes.length) {
-        // <label>text: <input /></label>
-        const formControlSelector =
-          'button, input, meter, output, progress, select, textarea'
-        const labelledFormControl = Array.from(
-          label.querySelectorAll(formControlSelector),
-        ).filter(element => element.matches(selector))[0]
-        if (labelledFormControl) elementsForLabel.push(labelledFormControl)
+  const matchingLabelledElements = Array.from(container.querySelectorAll('*'))
+    .filter(
+      element => element.labels || element.hasAttribute('aria-labelledby'),
+    )
+    .reduce((labelledElements, labelledElement) => {
+      const labelsId = labelledElement.getAttribute('aria-labelledby')
+        ? labelledElement.getAttribute('aria-labelledby').split(' ')
+        : []
+      const labelsValue = labelsId.length
+        ? labelsId.map(labelId => {
```

Comment:
What happens if an element had both a .labels property and the aria-labelledby? I don't know if that's valid or not.

Reply:
aria-labelledby has precedence: https://www.w3.org/TR/accname-1.1/#step2B

2B is for aria-labelledby, 2D for <label>

Reply:
In that case, I think this PR is good to go! Anyone have other thoughts?